data/lua/core/interface.lua

```diff
@@ -37,49 +37,48 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 		moving_unit:to_map(to_x, to_y)
 		wesnoth.wml_actions.redraw{}
 	end
-	---Fakes the move of a unit satisfying the given filter to position x, y.
-    ---Usable only during WML actions.
-	---Todo : maybe implement with C++.
+	---@param u unit
 	---@param to_x integer
-    ---@param to_y integer
-    ---@param u unit
+	---@param to_y integer
 	---@param queued boolean
-    function wesnoth.interface.move_unit_fake_queue(u, to_x, to_y, queued)
-        ---Fakes the move of a given unit, can be queued.
-		if to_x and to_y and u then
-			table.insert(wesnoth.interface.queued_movements, { to_x = to_x, to_y = to_y, u = u })
+	---@param scroll boolean
+	function wesnoth.interface.move_unit_fake_queue(u, to_x, to_y, queued, scroll)
+        if to_x and to_y and u then
+			if scroll == nil then
+				scroll = true
+			end
+			table.insert(queued_movements, { to_x = to_x, to_y = to_y, u = u, scroll = scroll })
 		end
 		if queued then
 			return
 		end
 
-        for _index, move in ipairs(wesnoth.interface.queued_movements) do
-            local moving_unit = move.u
-            local from_x, from_y = moving_unit.x, moving_unit.y
-
-            wesnoth.interface.scroll_to_hex(from_x, from_y)
-            to_x, to_y = wesnoth.paths.find_vacant_hex(move.to_x, move.to_y, moving_unit)
-
-            if to_x < from_x then
-                moving_unit.facing = "sw"
-            elseif to_x > from_x then
-                moving_unit.facing = "se"
-            end
-            moving_unit:extract()
-
+		for _index, move in ipairs(queued_movements) do
+			local moving_unit = move.u
+			local from_x, from_y = moving_unit.x, moving_unit.y
+			local scroll = move.scroll
+			if scroll then
+				wesnoth.interface.scroll_to_hex(from_x, from_y)
+			end
+			to_x, to_y = wesnoth.paths.find_vacant_hex(move.to_x, move.to_y, moving_unit)
+			if to_x < from_x then
+				moving_unit.facing = "sw"
+			elseif to_x > from_x then
+				moving_unit.facing = "se"
+			end
```

Comment:
I'm wondering about the logic here for setting the direction. It feels like this should be a place to use `wesnoth.map.get_relative_dir`, maybe? But I can't be sure since I'm not really sure what the intention is here.